Changed the statically initialized Subsystems to dynamic initialization

bradamiller opened the pull request:
Doing static creation of the subsystems will cause their constructors to run, potentially, before anything else in the library and could cause issues. There is no way to guarantee the order of static construction, so the safest way to do this is to make they dynamically allocated.

bradamiller committed:
Changed the statically initialized Subsystems to dynamic initialization

Doing static creation of the subsystems will cause their constructors to run, potentially, before anything else in the library and could cause issues. There is no way to guarantee the order of static construction, so the safest way to do this is to make they dynamically allocated.

calcmogul commented on src/include/Robot.h:
```diff
@@ -20,11 +20,11 @@
-  static Gripper gripper;
-  static Wrist wrist;
-  static Elevator elevator;
+  static OI * oi;
```
If you want to initialize these in RobotInit() instead, you should use `std::unique_ptr<OI>` for the type and `std::make_unique<OI>();` to initialize it.

calcmogul commented on src/cpp/Robot.cpp:
```diff
@@ -10,13 +10,13 @@
-Elevator Robot::elevator;
-
 void Robot::RobotInit() {
+	Robot::oi = new OI();
```
The comment about never using the `new` keyword in C++ anymore aside (see https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md#r-resource-management for more), this code doesn't compile. You need the lines you deleted above to provide definitions of the static variables. At the moment, you've only specified declarations, and the linker is throwing errors about undefined references.

If you really don't want the static definitions, provide functions like the following instead.
```
OI& Robot::OI() {
  static OI oi;
  return oi;
}
```
This is known as the construct-on-first-use idiom. It's how people generally work around the non-deterministic static construction order problem. https://isocpp.org/wiki/faq/ctors#static-init-order-on-first-use

calcmogul reviewed (changes requested):
There's also redundant includes added by your last commit, but that's already been merged, so...

bradamiller committed:
Add initializers for the static pointers

Missing initializers for the static pointers. Still need to install the CTRE vendor support to make it work.

bradamiller committed:
Switch to unique_ptr for commands and OI
